Schema Updates Implementation of Image and File Manipulation

mresong opened the pull request:
Updated schema with postgres

mresong committed:
Updated schema with postgres

elbzero commented on db/schema.rb:
```diff
@@ -13,6 +13,9 @@
 ActiveRecord::Schema.define(version: 20170726205721) do
 
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
```
Do we really need this?

mresong replied:
Eric told me to do it

elbzero commented:
This is not ready to go

mresong committed:
Merge remote-tracking branch 'remotes/origin/master' into rental-item-detail-start

mresong committed:
Implement rental item images with supportive frameworks

Image manipulation and handling (Carrierwave), S3 storage (fog-aws),
secure credential handling (figaro) heroku friendly.  Implement model
changes, and finally display on item page.

elbzero commented on app/assets/javascripts/application.js:
```diff
@@ -11,6 +11,7 @@
 // about supported directives.
 //
 //= require jquery
+
```
remove this blank line

elbzero commented on app/assets/stylesheets/scaffolds.scss:
```diff
@@ -0,0 +1,73 @@
+    font-size: 12px;
+    list-style: square;
+  }
+}
```
Usually this file is not necessary if we are using bootstrap. This automatically generated by rails. It can be removed.

elbzero commented on app/controllers/rental_items_controller.rb:
```diff
@@ -28,6 +30,9 @@ def create
       if @rental_item.save
+        params[:rental_item_images]['item_image'].each do |a|
+          @rental_item_image = @rental_item.rental_item_images.create!(:item_image => a)
+        end
```
Not sure how this library works but seems weird to assign a controller variable here in the loop, as it will just override the value each time. Unless if you just want the last value assigned for some reason.

elbzero commented on app/helpers/rental_item_images_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module RentalItemImagesHelper
+end
```
This is not needed, it is automatically generated

elbzero commented on test/controllers/rental_item_images_controller_test.rb:
```diff
@@ -0,0 +1,49 @@
+
+    assert_redirected_to rental_item_images_path
+  end
+end
```
We should get this hooked up to codeship so we can see if these tests are passing or not

mresong replied:
I agree!  Lets learn codeship hooking!

elbzero commented on test/models/rental_item_image_test.rb:
```diff
@@ -0,0 +1,7 @@
+class RentalItemImageTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Might be nice to put a small test here, instead of having nothing. Though could be a placeholder if we have nothing to test yet.

mresong replied:
For now i left this in as a placeholder... We can add a todo to make sure we dont forget

```suggestion
  # TODO Training of first test
  # end
```

elbzero reviewed (commented):
Just a handful of adjustments needed here

mresong committed:
Cleanup and create first Rspec test, then fix the test via TDD.

mresong committed:
Remove blank line

mresong committed:
Finish cleanup, tighten model requirements, added few tests, upgraded version

mresong committed:
Update test/models/rental_item_image_test.rb

Co-Authored-By: elbzero <elbzero@gmail.com>
